src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -138,6 +140,122 @@ type FSharpKernelBase () as this =
             context.Publish(CompletionsProduced(completionItems, requestCompletions))
         }
 
+    let handleRequestHoverText (requestHoverText: RequestHoverText) (context: KernelInvocationContext) =
+        async {
+            let! (parse, check, _ctx) = script.Value.Fsi.ParseAndCheckInteraction(requestHoverText.Code)
+
+            let res = FsAutoComplete.ParseAndCheckResults(parse, check, EntityCache())
+            let text = FSharp.Compiler.Text.SourceText.ofString requestHoverText.Code
+
+            // seem to be off by one
+            let line = requestHoverText.LinePosition.Line + 1
+            let col = requestHoverText.LinePosition.Character + 1
+
+            let fsiAssemblyRx = System.Text.RegularExpressions.Regex @"^\s*Assembly:\s+FSI-ASSEMBLY\s*$"
+
+            let lineContent = text.GetLineString(line - 1)
+            let! value =
+                async {
+                    match! res.TryGetSymbolUse (FSharp.Compiler.Range.mkPos line col) lineContent with
+                    | Ok (mine, _others) ->
+                        let fullName = 
+                            match mine with
+                            | FsAutoComplete.Patterns.SymbolUse.Val sym ->
+                                match sym.DeclaringEntity with
+                                | Some ent when ent.IsFSharpModule ->   
+                                    match ent.TryFullName with
+                                    | Some _ -> Some sym.FullName
+                                    | None -> None
+                                | _ -> None
+                            | _ ->
+                                None
+
+                        match fullName with
+                        | Some name ->
+                            let expr = if name.StartsWith "Stdin." then name.Substring 6 else name
+                            try return script.Value.Fsi.EvalExpression(expr) |> Some
```

Comment:
I'd like to understand more about how "safe" this evaluation is (i.e. not have side-effects)

Reply:
Since it just runs for module-bindings I figured they were already evaluated in the FSI-semantics?